packages/hub/node-tests/routes/prepaid-card-customizations-test.ts

```diff
@@ -0,0 +1,277 @@
+  return '';
+};
+
+const stubNonce = 'abc:123';
+let stubAuthToken = 'def--456';
+let stubTimestamp = process.hrtime.bigint();
+
+class StubAuthenticationUtils {
+  generateNonce() {
+    return stubNonce;
+  }
+  buildAuthToken() {
+    return stubAuthToken;
+  }
+  extractVerifiedTimestamp(_nonce: string) {
+    return stubTimestamp;
+  }
+
+  validateAuthToken(encryptedAuthToken: string) {
+    return handleValidateAuthToken(encryptedAuthToken);
+  }
+}
+
+let stubUserAddress = '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13';
+handleValidateAuthToken = function (encryptedString: string) {
+  expect(encryptedString).to.equal('abc123--def456--ghi789');
+  return stubUserAddress;
+};
+
+describe('POST /api/prepaid-card-customizations', function () {
+  let server: Server;
+  let db: Client;
+  let request: supertest.SuperTest<Test>;
+  let validPayload: any;
+  this.beforeEach(async function () {
+    let container!: Container;
+    server = await bootServerForTesting({
+      port: 3001,
+      registryCallback(registry: Registry) {
+        registry.register('authentication-utils', StubAuthenticationUtils);
+      },
+      containerCallback(serverContainer: Container) {
+        container = serverContainer;
+      },
+    });
+
+    let dbManager = await container.lookup('database-manager');
+    db = await dbManager.getClient();
+
+    let rows = [
+      ['AB70B8D5-95F5-4C20-997C-4DB9013B347C', 'https://example.com/a.svg', 'Pattern A'],
+      ['D2E94EA2-8124-44D8-B495-D3CF33D4C2A4', 'https://example.com/b.svg', 'Pattern B'],
+    ];
+    for (const row of rows) {
+      try {
+        await db.query('INSERT INTO prepaid_card_patterns(id, pattern_url, description) VALUES($1, $2, $3)', row);
+      } catch (e) {
+        console.error(e);
+      }
+    }
+    rows = [
+      ['C169F7FE-D83C-426C-805E-DF1D695C30F1', '#efefef', 'black', 'black', 'Solid Gray'],
+      [
+        '5058B874-CE21-4FC4-958C-B6641E1DC175',
+        'linear-gradient(139.27deg, #ff5050 16%, #ac00ff 100%)',
+        'white',
+        'white',
+        'Awesome Gradient',
+      ],
+    ];
+    for (const row of rows) {
+      try {
+        await db.query(
+          'INSERT INTO prepaid_card_color_schemes(id, background, pattern_color, text_color, description) VALUES($1, $2, $3, $4, $5)',
+          row
+        );
+      } catch (e) {
+        console.error(e);
+      }
+    }
+    validPayload = {
+      data: {
+        type: 'prepaid-card-customizations',
+        attributes: {
+          'issuer-name': 'Satoshi Nakamoto',
+        },
+        relationships: {
+          'color-scheme': {
+            data: {
+              type: 'prepaid-card-color-schemes',
+              id: '5058b874-ce21-4fc4-958c-b6641e1dc175',
+            },
+          },
+          pattern: {
+            data: {
+              type: 'prepaid-card-patterns',
+              id: 'ab70b8d5-95f5-4c20-997c-4db9013b347c',
+            },
+          },
+        },
+      },
+    };
```

Comment:
Is the structure/boilerplate to account for the possibility of using different sources for color schemes and patterns (eg. tables specific to external users) in the future? 

Wondering what the effect of a minimal request structure like below would be:
```
{
  type: 'prepaid-card-customizations'
  attributes: {
     'issuer-name': 'Satoshi Nakamoto',
     'color-scheme': '5058b874-ce21-4fc4-958c-b6641e1dc175',
     'pattern': 'ab70b8d5-95f5-4c20-997c-4db9013b347c'
   },
}
```

Reply:
This format follows the [JSON-API](https://jsonapi.org/) spec, which is the default serialization format for ember-data and orbit.js and what the card compiler API will use.